cmd/clusterctl/pkg/internal/util/yaml.go

```diff
@@ -52,11 +54,11 @@ func JoinYaml(yamls ...[]byte) []byte {
 // ToUnstructured takes a YAML and converts it to a list of Unstructured objects
 func ToUnstructured(rawyaml []byte) ([]unstructured.Unstructured, error) {
 	var ret []unstructured.Unstructured //nolint
-	for _, data := range bytes.Split(rawyaml, yamlSeparator) {
+	for _, data := range yamlSeparatorRegEx.Split(string(rawyaml), -1) {
```

Comment:
Instead of trying to brute force the separation, would it be better to use `YAMLReader` from `k8s.io/apimachiner.pkg/util/yaml` instead?

Reply:
done!